torch/package/package_exporter.py

```diff
@@ -929,10 +1016,36 @@ def _execute_dependency_graph(self):
                 if attrs.get("is_pickle") is True:
                     # This node came from save_source_pickle, we don't need to write any source for it.
                     continue
-
                 is_package = attrs["is_package"]
                 source = attrs["source"]
                 self._write_source_string(module_name, source, is_package)
+            elif action == _ModuleProviderAction.SELECTIVE_EXTERN:
+                is_package = attrs["is_package"]
+                source = attrs["source"]
+                selective_extern_template_file = str(
```

Comment:
I think this handling should be cleaned up. We don't need original source at all or the original_init at all, right?